### DISCUSSION TITLE
Расширение для Mozilla Firefox на Андроид

### DISCUSSION THREAD

**Discussion by Someone-G** [anticensority/runet-censorship-bypass#261] [Q&A]
Здравствуйте. Планируется ли портирование расширения для Mozilla Firefox на Андроид?

**Comment by ilyaigpetrov** [comment]
Там на Андроиде в FireFox не хватает реализации `proxy.settings`:
1) https://github.com/anticensority/runet-censorship-bypass/issues/86#issuecomment-1007658382
2) https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/proxy/settings#browser_compatibility

Один пользователь утверждал, что у него какое-то время работало, но перестало после обновления, в котором я убрал галочку публикации под эту ОС, т.к. версия под Андроид казалась мне нерабочей и могла помешать прохождению проверки человеком на addons.mozilla.org.

**Reply by Someone-G** [reply]
А в обход этой реализации никак не получится сделать? Ранее это расширение работало ведь...

**Reply by ilyaigpetrov** [reply]
Действительно, раньше могло работать. Отправил Мозилле версию 0.0.1.66 с поддержкой Андроида. Проверка может занять недели, но бывает и быстрее.

**Reply by glow12121** [reply]
Спасибо.

Вопрос дилетанта: а способа установить этот плагин на Хром на Андроиде нет, верно? Как я понимаю, там в принципе плагины невозможны.

**Reply by ilyaigpetrov** [reply]
В официальном Хроме под Андроид расширений нет. Есть его форки с поддержкой расширений, например, kiwi browser, но я ими не стал бы пользоваться.

**Reply by glow12121** [reply]
Понятно. Плагин под Фаерфокс был бы очень кстати.

**Reply by Someone-G** [reply]
> Вопрос дилетанта: а способа установить этот плагин на Хром на Андроиде нет, верно? Как я понимаю, там в принципе плагины невозможны.

Если только использовать форки, например, Kiwi. Он поддерживает расширения, в т.ч. и Обход Блокировок Рунета. Работает нормально.

**Reply by glow12121** [reply]
Жаль, что у Киви нет синхронизации, так бы его использовал.

**Reply by Someone-G** [reply]
> ...kiwi browser, но я ими не стал бы пользоваться.

Почему?

**Reply by ilyaigpetrov** [reply]
Просто не хватает доверия к команде этого браузера. Не хочу рисковать всеми паролями из менеджера паролей и расширениями-криптокошельками. Да и к утечке данных не готов, пусть даже собранной в маркетинговых целях.

**Reply by ilyaigpetrov** [reply]
0.0.1.66 под FireFox для Android [опубликована](https://addons.mozilla.org/en-US/firefox/addon/обход-блокировок-рунета/), но у меня пишет, что требуются `proxy.settings`.

Окно настроек после принятия соглашения автоматически не открывается. Открыть вручную можно через `menu ⋮ -> Extensions -> Выбрать расширение с синей лентой`.

**Reply by Someone-G** [reply]
> 0.0.1.66 под FireFox для Android [опубликована](https://addons.mozilla.org/en-US/firefox/addon/%D0%BE%D0%B1%D1%85%D0%BE%D0%B4-%D0%B1%D0%BB%D0%BE%D0%BA%D0%B8%D1%80%D0%BE%D0%B2%D0%BE%D0%BA-%D1%80%D1%83%D0%BD%D0%B5%D1%82%D0%B0/), но у меня пишет, что требуются `proxy.settings`.
> 
> Окно настроек после принятия соглашения автоматически не открывается. Открыть вручную можно через `menu ⋮ -> Extensions -> Выбрать расширение с синей лентой`.

Аналогично. :-(

**Reply by glow12121** [reply]
Почему-то параметр Хрома требуется:

![Screenshot_20240725-031240](https://github.com/user-attachments/assets/fdc3a0a6-e40a-474e-adc7-590dcee3d75c)

![Screenshot_20240725-031742](https://github.com/user-attachments/assets/427d82a5-cff8-49d0-85eb-14ba3f73a075)

**Reply by glow12121** [reply]
Если честно, очень рассчитываю на расширение О.Б.Р. для Фаерфокса. Специально перешел на этот браузер на Андроиде. Жаль, я не программист и не могу помочь.

**Comment by ilyaigpetrov** [comment]
Закрываю как дубликат https://github.com/anticensority/runet-censorship-bypass/issues/86 -- пишите туда, пожалуйста.
